tt_metal/hw/firmware/src/erisc.cc

```diff
@@ -18,7 +18,7 @@
 extern "C" {
 #endif
 
-void ApplicationHandler(void) __attribute__((__section__(".init")));
+  void ApplicationHandler(void);
```

Comment:
`ApplicationHandler()` is placed at `0x9040` which is start of `REGION_APP_CODE`

Currently it is getting placed there by virtue of `void __attribute__((section("erisc_l1_code.0"), naked)) ApplicationHandler(void)`

Removing .init from the prototype should be ok. It was not effective anyway the way `erisc-b0-app-sections.ld` is setup.